src/java.base/share/classes/sun/util/locale/provider/CalendarNameProviderImpl.java

```diff
@@ -189,7 +189,14 @@ public String getDisplayNameImpl(String calendarType, int field, int value, int
                             if (name.isEmpty()) {
                                 continue;
                             }
-                            map.put(name, base + i);
+                            if (field == AM_PM && !javatime && i > PM) {
+                                // when dealing with calendar fields, don't set AM_PM field value
```

Comment:
Make it explicit that this block only serves for `java.util.Calendar`, not `java.time.format.DateTimeFormatter(Builder)`.

Reply:
Hello Naoto, I've now updated the PR to be more explicit in this code comment.